crates/matrix-sdk-sled/src/state_store.rs

```diff
@@ -299,24 +313,49 @@ impl SledStore {
         &self,
         event: &[u8],
     ) -> Result<T, SledStoreError> {
-        if let Some(key) = &*self.store_key {
-            let encrypted: EncryptedEvent = serde_json::from_slice(event)?;
-            Ok(key.decrypt(encrypted)?)
+        if let Some(key) = &*self.store_cipher {
+            Ok(key.decrypt_value(event)?)
         } else {
             Ok(serde_json::from_slice(event)?)
         }
     }
 
-    pub async fn save_filter(&self, filter_name: &str, filter_id: &str) -> Result<()> {
-        self.session.insert(("filter", filter_name).encode(), filter_id)?;
+    fn encode_key<T: EncodeSecureKey + EncodeKey + ?Sized>(
+        &self,
+        table_name: &str,
+        key: &T,
+    ) -> Vec<u8> {
+        if let Some(store_cipher) = &*self.store_cipher {
+            key.encode_secure(table_name, store_cipher).to_vec()
+        } else {
+            key.encode()
+        }
+    }
+
+    fn encode_key_with_counter<A: EncodeSecureKey + EncodeKey + ?Sized>(
```

Comment:
Should `EncodeSecureKey` have `EncodeKey` as a supertrait?

Reply:
`EncodeSecureKey` already existed in here for the cryptostore and I attempted to keep the changes to that module as limited as possible. Arguably, as in `indexeddb` shown, these could very well be merged as they are both needed in all cases anyways. Would like to know what @poljar thinks about that.

Reply:
Yeah I considered just merging them, I don't think there's a good reason why I didn't do that so feel free to do so.